dashboards-observability/public/components/visualizations/charts/pie/pie.tsx

```diff
@@ -17,42 +17,42 @@ export const Pie = ({ visualizations, layout, config }: any) => {
   } = visualizations.data.rawVizData;
   const { defaultAxes } = visualizations.data;
   const { dataConfig = {}, layoutConfig = {} } = visualizations?.data?.userConfigs;
-  const xaxis = visualizations.data?.rawVizData?.pie?.dataConfig?.dimensions 
-    ? visualizations.data?.rawVizData?.pie?.dataConfig?.dimensions.filter((item)  => item.label)
+  const xaxis = visualizations.data?.rawVizData?.pie?.dataConfig?.dimensions
+    ? visualizations.data?.rawVizData?.pie?.dataConfig?.dimensions.filter((item) => item.label)
     : [];
   const yaxis = visualizations.data?.rawVizData?.pie?.dataConfig?.metrics
-    ? visualizations.data?.rawVizData?.pie?.dataConfig?.metrics.filter((item)  => item.label)
+    ? visualizations.data?.rawVizData?.pie?.dataConfig?.metrics.filter((item) => item.label)
     : [];
   const type = dataConfig?.chartStyles?.mode ? dataConfig?.chartStyles?.mode[0]?.modeId : 'pie';
   const lastIndex = fields.length - 1;
   const colorTheme = dataConfig?.chartStyles?.colorTheme
     ? dataConfig?.chartStyles?.colorTheme
     : { name: DEFAULT_PALETTE };
-  const showLegend = dataConfig?.legend?.showLegend === 'hidden' ? false : vis.showLegend;
-  const legendPosition = dataConfig?.legend?.position || vis.legendPosition;
+  const showLegend = dataConfig?.legend?.showLegend === 'hidden' ? false : vis.showlegend;
+  const legendPosition = dataConfig?.legend?.position || vis.legendposition;
   const legendSize = dataConfig?.legend?.size || vis.legendSize;
   const labelSize = dataConfig?.chartStyles?.labelSize || vis.labelSize;
 
   if (isEmpty(xaxis) || isEmpty(yaxis))
-    return <EmptyPlaceholder icon={visualizations?.vis?.iconType} />;
+    return <EmptyPlaceholder icon={visualizations?.vis?.icontype} />;
 
-  let valueSeries;
-  if (!isEmpty(xaxis) && !isEmpty(yaxis)) {
-    valueSeries = [...yaxis];
-  } else {
-    valueSeries = defaultAxes.yaxis || take(fields, lastIndex > 0 ? lastIndex : 1);
-  }
+    let valueSeries;
+    if (!isEmpty(xaxis) && !isEmpty(yaxis)) {
+      valueSeries = [...yaxis];
+    } else {
+      valueSeries = defaultAxes.yaxis || take(fields, lastIndex > 0 ? lastIndex : 1);
+    }
 
   const invertHex = (hex: string) =>
     (Number(`0x1${hex}`) ^ HEX_CONTRAST_COLOR).toString(16).substr(1).toUpperCase();
 
-  const createLegendLabels = (dimLabels: string[], xaxisLables:string[] ) => {
-    return dimLabels.map((label:string, index:number) => {
+  const createLegendLabels = (dimLabels: string[], xaxisLables: string[]) => {
+    return dimLabels.map((label: string, index: number) => {
       return [xaxisLables[index], label].join(',');
     });
   };
-  
-  const labelsOfXAxis = () => {
+
+  const labelsOfXAxis = useMemo(() => {
```

Comment:
do we need useMemo here.

Reply:
As we have utilized the createLegendLabels method to map Legend labels. That should be updated only on data and axis change. So we have created memoize value here.